python/hsfs/core/feature_view_engine.py

```diff
@@ -183,6 +189,93 @@ def delete(self, name, version=None):
         else:
             return self._feature_view_api.delete_by_name(name)
 
+    def get_schema(self, feature_view: FeatureView, training_dataset_version: int):
+        """
+        Function that returns the schema of the training dataset generated using the feature view.
+
+        # Arguments
+            feature_view: `FeatureView`. The feature view for which the schema is to be generated.
+            training_dataset_version: `int`. Version of the training dataset for which the schema is to be generated.
+
+        # Returns
+            `List[training_dataset_feature.TrainingDatasetFeature]`: List of training dataset features objects.
+
+        # Raises
+            `ValueError` if the  training dataset version provided cannot be found.
+        """
+
+        if not feature_view.transformation_functions:
+            # Return the features in the feature view if the no transformation functions in the feature view.
+            return feature_view.features
+        else:
+            # Create schema based on transformation function.
+            (
+                transformed_feature_names,
+                output_types_features,
+                transformed_label_names,
+                output_types_labels,
```

Comment:
no sure i got why do it like this?
wouldn't it be the same using something like this:
```
transformed_names = []
output_types = []
```
Aka. no separating features and lables, since they are joined either way?
maybe even just having list of `training_dataset_feature.TrainingDatasetFeature` and adding to it directly?

Reply:
Yes you I right I have updated the code based on the suggestion.

Reply:
why not just work with a list of `training_dataset_feature.TrainingDatasetFeature`? i think it would be cleaner, right now you have `transformed_features` and `transformed_labels`, this make you have multiple loops to construct list of `training_dataset_feature.TrainingDatasetFeature`.

Reply:
The schema of the transformed feature follows an order. Any training dataset or data read from the feature view will contains all the untransformed_feature followed by the transformed_features. 
So the `training_dataset_schema` must return a list that has features in the order  untransformed_featrures, transformed_features, labels and transformed_labels. 

To preserve the order, I think if we use single list we would need two for loops one for labels and the other for normal features. That was why I did the changes using two lists instead of two for loops, but do you think its better and cleaner to use two for loops instead of two lists.